cloud/gke.go

```diff
@@ -0,0 +1,1080 @@
+			NodeCount: cc.NodeCount,
+		}).Context(context.Background()).Do()
+		if err != nil {
+			return nil, err
+		}
+		banzaiUtils.LogInfof(banzaiConstants.TagUpdateCluster, "Nodepool %s size change is called for project %s, zone %s and cluster %s. Status Code %v", cc.NodePoolID, cc.ProjectID, cc.Zone, cc.Name, updateCall.HTTPStatusCode)
+		if updatedCluster, err = waitForCluster(svc, cc); err != nil {
+			banzaiUtils.LogError(banzaiConstants.TagUpdateCluster, "Contains error", err)
+			return nil, err
+		}
+	}
+	return updatedCluster, nil
+}
+
+func waitForNodePool(svc *gke.Service, cc GKECluster) error {
+	const TAG = "waitForNodePool"
+	message := ""
+	for {
+		nodepool, err := svc.Projects.Zones.Clusters.NodePools.Get(cc.ProjectID, cc.Zone, cc.Name, cc.NodePoolID).Context(context.TODO()).Do()
+		if err != nil {
+			return err
+		}
+		if nodepool.Status == statusRunning {
+			banzaiUtils.LogInfof(TAG, "Nodepool %v is running", cc.Name)
+			return nil
+		}
+		if nodepool.Status != message {
+			banzaiUtils.LogInfof(TAG, "%v nodepool %v", string(nodepool.Status), cc.NodePoolID)
+			message = nodepool.Status
+		}
+		time.Sleep(time.Second * 5)
+	}
+}
+
+func DeleteGoogleCluster(cs *banzaiSimpleTypes.ClusterSimple, c *gin.Context) bool {
+
+	banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Start delete google cluster")
+
+	if cs == nil {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "<nil> cluster")
+		return false
+	}
+
+	// set google props
+	banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Load Google props from database")
+	database.SelectFirstWhere(&cs.Google, banzaiSimpleTypes.GoogleClusterSimple{ClusterSimpleId: cs.ID})
+	gkec := GKECluster{
+		ProjectID: cs.Google.Project,
+		Name:      cs.Name,
+		Zone:      cs.Location,
+	}
+
+	if deleteCluster(&gkec, c) {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Delete succeeded")
+		return true
+	} else {
+		banzaiUtils.LogWarn(banzaiConstants.TagGetCluster, "Can't delete cluster from cloud!")
+		SetResponseBodyJson(c, http.StatusBadRequest, gin.H{
+			JsonKeyStatus:     http.StatusBadRequest,
+			JsonKeyMessage:    "Can't delete cluster!",
+			JsonKeyResourceId: cs.ID,
+		})
+		return false
+	}
+}
+
+func deleteCluster(cc *GKECluster, c *gin.Context) bool {
+
+	banzaiUtils.LogInfo(banzaiConstants.TagDeleteCluster, "Get Google Service Client")
+
+	svc, err := GetGoogleServiceClient()
+	if err != nil {
+		SetResponseBodyJson(c, http.StatusInternalServerError, gin.H{
+			JsonKeyStatus:  http.StatusInternalServerError,
+			JsonKeyMessage: err,
+		})
+		return false
+	}
+	banzaiUtils.LogInfo(banzaiConstants.TagDeleteCluster, "Get Google Service Client succeeded")
+
+	banzaiUtils.LogInfof(banzaiConstants.TagDeleteCluster, "Removing cluster %v from project %v, zone %v", cc.Name, cc.ProjectID, cc.Zone)
+	deleteCall, err := svc.Projects.Zones.Clusters.Delete(cc.ProjectID, cc.Zone, cc.Name).Context(context.Background()).Do()
```

Comment:
I think this can be replaced with function GetClusterGoogle()

Reply:
It's the Delete function, not Get

Reply:
Ohh okey, you are right sorry my mistake:)